gpu/src/resources.rs

```diff
@@ -56,14 +57,26 @@ impl FilesystemResourceLoader {
 }
 
 impl ResourceLoader for FilesystemResourceLoader {
-    fn slurp(&self, virtual_path: &str) -> Result<Vec<u8>, IOError> {
+    fn slurp(&self, virtual_path: &str) -> Result<Cow<'static, [u8]>, IOError> {
         let mut path = self.directory.clone();
         virtual_path
             .split('/')
             .for_each(|segment| path.push(segment));
 
         let mut data = vec![];
         File::open(&path)?.read_to_end(&mut data)?;
-        Ok(data)
+        Ok(data.into())
+    }
+}
+
+pub struct EmbeddedResourceLoader;
```

Comment:
This can probably use some tuning in the future to only include the files actually needed. But it works for now.

Reply:
Won't this cause a rebuild when changing the shaders? One of the benefits of the `FilesystemResourceLoader` is live shader editing without rebuilding. I know it's not possible to use the filesystem in WebAssembly, but maybe we should make this enabled on the wasm target only.

Reply:
I moved the `EmbeddedResourceLoader` into pathfinder_resources and added a bunch of feature flags to control what gets included.
The demo runs.